src/main/scala/rise/autotune/configFileGeneration.scala

```diff
@@ -299,4 +300,35 @@ object configFileGeneration {
     output.toSeq.sortBy(_.name)
   }
 
+  // function to check cycle in parameter dependencies
+  def check_no_cycle(distribution: Map[NatIdentifier,
+    (Set[constraints.Constraint], Set[NatIdentifier])]
+                    ): Boolean = {
+
+    def check_no_cycle_rec(param: NatIdentifier,
+                           dependencies: Set[NatIdentifier],
+                           distribution: Map[NatIdentifier,
+                             (Set[constraints.Constraint], Set[NatIdentifier])],
+                           visited: Set[NatIdentifier]
+                          ): Boolean = {
+
+      dependencies.size match {
+        case 0 => true
+        case _ => dependencies.exists(dependency => {
+          dependency.name.equals(param.name) || visited.contains(dependency)
```

Comment:
You could avoid testing `dependency.name.equals(param.name)` by adding `param` to `visited` in the beginning.